libraries/metric/lib/util/diagnostics.ts

```diff
@@ -50,3 +50,10 @@ export const seriesDistributionTypeNotRegistered = (
 
 export const shouldNeverHappen = (bugLocation: string) =>
   `This should never happen.\nThere is likely a bug in the ${bugLocation}.`;
+
+export class UndefinedValueError extends Error {
```

Comment:
Unless it is a type of error that can be used by multiple different classes or is more of a base class, it would be better to keep the errors within the files that throw it.